src/main/java/com/onaio/steps/tasks/RootDetectionTask.java

```diff
@@ -39,7 +39,13 @@ public void onPostExecute(Boolean isRooted) {
         }
     }
 
-    public void deleteAllData() {
+    public void deleteAllData(Context context, DatabaseHelper db) {
+
+        if (context instanceof Activity) {
```

Comment:
What happens when this is called from the BootReceiver directly? I blelieve that this context will not be an instance of the Activity and therefore the username and password can be extracted from the device

Reply:
yes, the context will not be the activity context, I added this because in [KeyValueStore](https://github.com/onaio/steps-app/blob/master/src/main/java/com/onaio/steps/helper/KeyValueStore.java#L46) we are getting the preferences by using an Activity. What would you suggest, do we modify the KeyValueStore and get the preferences by using the `Context` instead of `Activity`.

Reply:
So that the prefs can be deleted even after booting into a rooted device

Reply:
done